File: clients/python/model/colored_vertex.py
from .vec2_float import Vec2Float
from .color_float import ColorFloat
class ColoredVertex:
    def __init__(self, position, color):
        self.position = position
        self.color = color
    @staticmethod
    def read_from(stream):
        position = Vec2Float.read_from(stream)
        color = ColorFloat.read_from(stream)
        return ColoredVertex(position, color)
    def write_to(self, stream):
        self.position.write_to(stream)
        self.color.write_to(stream)
    def __repr__(self):
        return "ColoredVertex(" + \
            repr(self.position) + "," + \
            repr(self.color) + \
            ")"

File: clients/javascript/model/colored-vertex.js
const Vec2Float = require('./vec2-float').Vec2Float;
const ColorFloat = require('./color-float').ColorFloat;
class ColoredVertex {
    constructor(position, color) {
        this.position = position;
        this.color = color;
    }
    static async readFrom(stream) {
        let position;
        position = await Vec2Float.readFrom(stream);
        let color;
        color = await ColorFloat.readFrom(stream);
        return new ColoredVertex(position, color);
    }
    async writeTo(stream) {
        let position = this.position;
        await position.writeTo(stream);
        let color = this.color;
        await color.writeTo(stream);
    }
}
module.exports = { ColoredVertex: ColoredVertex }
